File: LeetCode/Python/Easy/1704-Determine-if-String-Halves-Are-Alike/solution.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        n=len(s)

        half=n // 2

        first_half_vowels=0

        for index in range(0, half):
            if self.is_vowel(s[index]):
                first_half_vowels+=1

        second_half_vowels=0

        for index in range(half, n):
            if self.is_vowel(s[index]):
                second_half_vowels+=1

        return first_half_vowels==second_half_vowels

    def is_vowel(self, character: chr) -> bool:
        vowels=set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'])

        return character in vowels

File: LeetCode/Rust/Easy/1704-Determine-if-String-Halves-Are-Alike/solution.rs
use std::collections::{BTreeSet};

impl Solution
{
    pub fn halves_are_alike(s: String) -> bool
    {
        let n: usize=s.len();

        let half: usize=n / 2;

        let vowels: Vec<char>=vec!['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];

        let mut vowels_set: BTreeSet<char>=vowels.into_iter().collect::<BTreeSet<char>>();

        let mut first_half_vowels: usize=0;

        let mut second_half_vowels: usize=0;

        for (index, character) in s.chars().enumerate()
        {
            if vowels_set.contains(&character)
            {
                if index < half
                {
                    first_half_vowels+=1;
                }
                else
                {
                    second_half_vowels+=1;
                }
            }
        }

        first_half_vowels==second_half_vowels
    }
}